From d2faf531d733cb4a5d559f015b965e64619eae99 Mon Sep 17 00:00:00 2001
From: RubenKelevra <ruben@vfn-nrw.de>
Date: Sat, 14 May 2016 17:52:13 +0200
Subject: [PATCH 1/1] ipq806x: Add Archer C2600 to image Makefile

Signed-off-by: Josh Bendavid <joshbendavid@gmail.com>
[darryl@sokoloski.ca: submitting patches on behalf of Josh Bendavid]
Signed-off-by: Darryl Sokoloski <darryl@sokoloski.ca>

source: https://patchwork.ozlabs.org/patch/612487/
---
 target/linux/ipq806x/image/Makefile | 29 ++++++++++++++++++++++++++++-
 1 file changed, 28 insertions(+), 1 deletion(-)

diff --git a/target/linux/ipq806x/image/Makefile b/target/linux/ipq806x/image/Makefile
index ff8b2c7..a5fc836 100644
--- a/target/linux/ipq806x/image/Makefile
+++ b/target/linux/ipq806x/image/Makefile
@@ -36,6 +36,11 @@ define Build/append-dtb
 	cat $(DTS_DIR)/$(DEVICE_DTS).dtb >> $@
 endef
 
+define Build/append-new-dtb
+       $(call Image/BuildDTB,../dts/$(DEVICE_DTS).dts,$@.dtb)
+       cat $@.dtb >> $@
+endef
+
 define Build/append-file
 	cat $(1) >> $@
 endef
@@ -86,6 +91,19 @@ define Device/DniImage
 endef
 DEVICE_VARS += KERNEL_SIZE NETGEAR_BOARD_ID NETGEAR_HW_ID DEVICE_BLOCK_SIZE DEVICE_PAGE_SIZE
 
+define Device/TpSafeImage
+	PROFILES += $$(DEVICE_NAME)
+	FILESYSTEMS := squashfs
+	KERNEL_SUFFIX := -uImage
+	KERNEL = kernel-bin | append-new-dtb | uImage none
+	KERNEL_NAME := zImage
+	TPLINK_BOARD_ID :=
+	IMAGES := factory.bin sysupgrade.bin
+	IMAGE/factory.bin := tplink-safe factory
+	IMAGE/sysupgrade.bin := tplink-safe sysupgrade
+endef
+DEVICE_VARS += TPLINK_BOARD_ID
+
 define Device/AP148
 	$(call Device/FitImage)
 	$(call Device/UbiFit)
@@ -104,6 +122,15 @@ define Device/AP148-legacy
 	BOARD_NAME := ap148
 endef
 
+define Device/C2600
+	$(call Device/TpSafeImage)
+	DEVICE_DTS := qcom-ipq8064-c2600
+	BLOCKSIZE := 128KiB
+	PAGESIZE := 2048
+	BOARD_NAME := c2600
+	TPLINK_BOARD_ID := C2600
+endef
+
 define Device/D7800
 	$(call Device/DniImage)
 	DEVICE_DTS := qcom-ipq8064-d7800
@@ -133,6 +160,6 @@ define Device/R7500
 	BOARD_NAME := r7500
 endef
 
-TARGET_DEVICES += AP148 AP148-legacy D7800 DB149 R7500
+TARGET_DEVICES += AP148 AP148-legacy C2600 D7800 DB149 R7500
 
 $(eval $(call BuildImage))
-- 
2.8.2

